presto-main/src/main/java/com/facebook/presto/sql/planner/EqualityInference.java

```diff
@@ -269,7 +270,8 @@ Expression getScopedCanonical(Expression expression, Predicate<Symbol> symbolSco
             expression = normalizeInPredicateToEquality(expression);
             if (expression instanceof ComparisonExpression &&
                     isDeterministic(expression) &&
-                    !mayReturnNullOnNonNullInput(expression)) {
+                    !mayReturnNullOnNonNullInput(expression) &&
+                    !isDynamicFilter(expression)) {
```

Comment:
Why do we need it here?

Reply:
looks like you added it when Dynamic filter was comparision expression. Not required now. has been removed.